client/ayon_core/plugins/load/delete_old_versions.py

```diff
@@ -1,501 +1,478 @@
+
+    def main(self, data, remove_publish_folder):
+        # Size of files.
+        size = 0
+        if not data:
+            return size
+
+        if remove_publish_folder:
+            size = self.delete_whole_dir_paths(data["dir_paths"].values())
+        else:
+            size = self.delete_only_repre_files(
+                data["dir_paths"], data["file_paths_by_dir"]
+            )
+
+        for version in data["versions"]:
+            orig_version_tags = version["data"].get("tags") or []
+            version_tags = [tag for tag in orig_version_tags]
+            if "deleted" not in version_tags:
+                version_tags.append("deleted")
+
+            if version_tags == orig_version_tags:
+                continue
+
+        self._ftrack_delete_versions(data)
+
+        return size
+
+    def _ftrack_delete_versions(self, data):
+        """Delete version on ftrack.
+
+        Handling of ftrack logic in this plugin is not ideal. But in OP3 it is
+        almost impossible to solve the issue other way.
+
+        Note:
+            Asset versions on ftrack are not deleted but marked as
+                "not published" which cause that they're invisible.
+
+        Args:
+            data (dict): Data sent to product loader with full context.
+        """
+
+        # First check for ftrack id on folder entity
+        #   - skip if ther is none
+        ftrack_id = data["folder"]["attrib"].get("ftrackId")
+        if not ftrack_id:
+            self.log.info((
+                "Folder does not have filled ftrack id. Skipped delete"
+                " of ftrack version."
+            ))
+            return
+
+        # Check if ftrack module is enabled
+        addons_manager = AddonsManager()
+        ftrack_addon = addons_manager.get("ftrack")
+        if not ftrack_addon or not ftrack_addon.enabled:
+            return
+
+        import ftrack_api
+
+        session = ftrack_api.Session()
+        product_name = data["product"]["name"]
+        versions = {
+            '"{}"'.format(version_doc["name"])
+            for version_doc in data["versions"]
+        }
+        asset_versions = session.query(
+            (
+                "select id, is_published from AssetVersion where"
+                " asset.parent.id is \"{}\""
+                " and asset.name is \"{}\""
+                " and version in ({})"
+            ).format(
+                ftrack_id,
+                product_name,
+                ",".join(versions)
+            )
+        ).all()
+
+        # Set attribute `is_published` to `False` on ftrack AssetVersions
+        for asset_version in asset_versions:
+            asset_version["is_published"] = False
+
+        try:
+            session.commit()
+
+        except Exception:
+            msg = (
+                "Could not set `is_published` attribute to `False`"
+                " for selected AssetVersions."
+            )
+            self.log.error(msg)
+            self.message(msg)
```

Comment:
Should this PR include Ftrack addon specific testing? Which may be hard to do as long as the Ftrack addon may not be fully up-to-date with the refactored `ayon-core`?

Reply:
> Should this PR include Ftrack addon specific testing? Which may be hard to do as long as the Ftrack addon may not be fully up-to-date with the refactored `ayon-core`?
@iLLiCiTiT

Reply:
Ha, I didn't know it's there is ftrack removal. Should be removed.